csharp/ql/lib/semmle/code/csharp/exprs/Call.qll

```diff
@@ -76,17 +76,24 @@ class Call extends DotNet::Call, Expr, @call {
    * ```
    */
   cached
-  override Expr getArgumentForParameter(DotNet::Parameter p) {
-    this.getTarget().getAParameter() = p and
+  final override Expr getArgumentForParameter(DotNet::Parameter p) {
+    result = this.getArgumentForParameterInternal(this.getTarget(), p)
+  }
+
+  /** INTERNAL: Do not use. */
+  Expr getArgumentForParameterInternal(Callable c, DotNet::Parameter p) {
+    c.getAParameter() = p and
     (
       // Appears in the positional part of the call
-      result = this.getImplicitArgument(p.getPosition()) and
+      result = this.getImplicitArgument(p) and
       (
         p.(Parameter).isParams()
         implies
         (
           isValidExplicitParamsType(p, result.getType()) and
-          not this.hasMultipleParamsArguments()
+          not this.hasMultipleParamsArguments(c)
+          or
+          result.getType().isImplicitlyConvertibleTo(p.getType().(ArrayType).getElementType())
```

Comment:
Why is this needed?

Reply:
@hvitved raised the point below that `isImplicitlyConvertibleTo` might not work for generics. This is something that we should check and fix here. `final override Expr getArgumentForParameter` should work for traced DBs.